pkg/controller/proxyconfig/status.go

```diff
@@ -0,0 +1,100 @@
+package proxyconfig
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// syncProxyStatus...
+func (r *ReconcileProxyConfig) syncProxyStatus(proxy *configv1.Proxy, infra *configv1.Infrastructure, network *configv1.Network, cluster *corev1.ConfigMap) error {
+	updated := proxy.DeepCopy()
+
+	apiServerURL, err := url.Parse(infra.Status.APIServerURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse API server URL")
+	}
+	internalAPIServer, err := url.Parse(infra.Status.APIServerInternalURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse API server internal URL")
+	}
+
+	set := sets.NewString(
+		"127.0.0.1",
+		"localhost",
+		network.Status.ServiceNetwork[0],
+		apiServerURL.Hostname(),
+		internalAPIServer.Hostname(),
+	)
+	platform := infra.Status.PlatformStatus.Type
+
+	// TODO: Does a better way exist to get machineCIDR and controlplane replicas?
```

Comment:
@squeed @danwinship PTAL and let me know if you have a better approach for parsing `replicas` and `machineCIDR` from the installer config or using some other resource to get the data.